### DISCUSSION TITLE
Copilot fails to sign-in to github

### DISCUSSION THREAD

**Discussion by gaashh** [community/community#39166] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I try to sign in from IntelliJ (2022.2.3/Ultimate) using Copilot plugin (1.1.35.2063). 

"Sign in to GitHub" opens a browser page and I reach the "Congratulations, you're all set!
Your device is now connected." message. But at the IntelliJ side the plugin dialog box closes and it does not sign in.

**Comment by iamcjwilliams** [comment]
@gaashh Have you tried logging out and logging back in? What are the steps taken to receive the error mentioned?

**Reply by gaashh** [reply]
@iamcjwilliams 

Logout and login did not help.

The process is:
1. Click the copilot icon at InelliJ bottom toolbar
2. "Login to github"
3. "Sign in to GitHub" dialog popup. Click "Copy and Open"
4. At browser "Device Activation" page opens. Paste the device code. 
5. Click "Authorize GitHub Copilot plugin" that leads to "Congratulations, you're all set! Your device is now connected."
6. At the IntelliJ side "Waiting for GitHub Authentication" window pops up. After a short while it disappears. 
7.

**Reply by coderGambit** [reply]
I have this exact issue.

**Reply by parunev** [reply]
Any updates?

**Reply by unax26203** [reply]
Did u solve this issue??

**Reply by lironzizva** [reply]
Same here.

**Reply by misha21742** [reply]
same here

**Reply by brajeshmehra24** [reply]
I'm also getting the same issue in Goland

**Reply by cwyspy100** [reply]
same here pycharm 2022.2， a  month cannot login in

**Reply by cwyspy100** [reply]
i can login github，but in idea or pycharm vscode cannot login in for reason Extension activation failed: "connect ETIMEDOUT 192.30.255.117:443" ， why。。。

**Comment by iamcjwilliams** [comment]
@gaashh Our copilot team will review this issue

**Comment by gaashh** [comment]
@iamcjwilliams any update?

**Reply by tilmann1** [reply]
@gaashh Please open a support ticket via https://support.github.com/contact/bug-report, thanks!

**Comment by rkishony** [comment]
same issue for me. is it related to institutional repos perhaps?

**Reply by iamcjwilliams** [reply]
@rkishony Have you tried troubleshooting your issue here: https://docs.github.com/en/copilot/troubleshooting-github-copilot

**Comment by RobertronS** [comment]
I have the same issue, is there any fix?

**Reply by iamcjwilliams** [reply]
@RobertronS can you provide a screenshot of the issue you're receiving and the steps to replicate it?

**Reply by Brainshack** [reply]
I get the same issue. You start the login from the IDE, you enter the code, it says "Congratulations, you're all set!
Your device is now connected." And then in the idea, when I try to refresh copilot it says i am not logged in.

I checked the idea.log file and this is what appears:

"2023-02-09 20:52:10,709 [ 220766]   INFO - #copilot - [INFO] [auth] [2023-02-09T19:52:10.707Z] Invalid copilot token: missing token: 403 
2023-02-09 20:52:27,021 [ 237078]   INFO - #copilot - [INFO] [auth] [2023-02-09T19:52:27.020Z] Invalid copilot token: missing token: 403 "

**Reply by Brainshack** [reply]
Ok I found the issue: apparently i wasnt done signing up for copilot. For me it felt like I completed it, after entering my payment info. But there was a step more to complete it.

**Reply by evgfilim1** [reply]
I'm getting the same issue as @Brainshack.
After logging in and seeing "Congratulations, you're all set" on GitHub page, loading popup from Copilot plugin disappear and I'm still not logged in.
Also one more weird thing: after restarting my IDE, Copilot shows a (useless?) popup: "Contact support", no more info at all.
Checking logs doesn't help much, it says "Invalid copilot token: missing token: 403".

Also, if it's important, I'm using PyCharm.

**Reply by cfouf** [reply]
> I'm getting the same issue as @Brainshack. After logging in and seeing "Congratulations, you're all set" on GitHub page, loading popup from Copilot plugin disappear and I'm still not logged in. Also one more weird thing: after restarting my IDE, Copilot shows a (useless?) popup: "Contact support", no more info at all. Checking logs doesn't help much, it says "Invalid copilot token: missing token: 403".
> 
> Also, if it's important, I'm using PyCharm.

Same. Happened after I installed copilot on my second computer or after the update (ain't really knowing the cause, both were done at the same time)

**Reply by ihordyrman** [reply]
Same issue here. I can't log in from IntelliJ and VSCode.
```
[INFO] [auth] [2023-02-16T07:05:27.641Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2023-02-16T07:05:27.644Z] GitHub Copilot could not connect to server. Extension activation failed: "Contact Support"
```

**Reply by napalmpapalam** [reply]
same behaviour for me in WebStorm

**Reply by dm-mrtnvch** [reply]
the same issue. webstorm. can't link my ide and copilot.

**Reply by shibasis0801** [reply]
same issue with IDEA 2022.2.3

**Reply by nortain** [reply]
Same issue with IntelliJ IDEA 2022.3 (Ultimate Edition)

**Reply by shibasis0801** [reply]
<img width="527" alt="image" src="https://user-images.githubusercontent.com/13601918/220294461-c941d809-351d-44de-93b3-665742590325.png">

Getting stuck at this step

@iamcjwilliams

**Reply by iamcjwilliams** [reply]
@shibasis0801 Are you getting prompted to enter an authentication in your browser?

**Reply by shibasis0801** [reply]
Thank you for the time
Yes I am getting a prompt and 
![image](https://user-images.githubusercontent.com/13601918/220831684-db92829a-1cf3-4c55-af18-af879d4c1bad.png)

It is successful
![image](https://user-images.githubusercontent.com/13601918/220831781-8e128a2a-387b-46a4-932b-0b0e8a258611.png)

But after that waiting dialog still stays
I didn't get anything interesting in the logs

Also, I am able to use it in VS Code so this doesn't seem like a network issue right ? 
Didn't do anything specific to configure for VS Code. 
( This is my work laptop in a corporate network )

Please let me know if any details are needed 

IDEA Version -> IntelliJ Ultimate 2022.3.2
Plugin Version -> 1.2.3.2385

**Reply by shibasis0801** [reply]
@iamcjwilliams Could you please let me know?

**Reply by iamcjwilliams** [reply]
@shibasis0801 IT admins put restrictions on their corporate networks. Therefore, it's imposed across ALL network traffic. Copilot just requires network access, so it falls inside the corporate web traffic policy. Have you tried using copilot outside of your corporate network

**Reply by AndersWinther-Dahl** [reply]
> @shibasis0801 IT admins put restrictions on their corporate networks. Therefore, it's imposed across ALL network traffic. Copilot just requires network access, so it falls inside the corporate web traffic policy. Have you tried using copilot outside of your corporate network

Is it possible to confirm if that's the case?
Last time I looked in the idea.log file it wasn't clear to me, if it's caused by network or other policy restriction. 

Maybe the plugin creator should log some more or give a better explanation in the error-pop-up.
Just a suggestion 😊

**Reply by shibasis0801** [reply]
@iamcjwilliams 
It's working fine on my personal laptop. I guessed it would be some kind of corp restriction on the Work system. 
But I am able to use VS code with copilot on the work laptop. 
If there is a solution for this then please let me know. 

Alternatively I guess you would be storing the auth token somewhere locally in the plugin to persist across restarts.
Could you please let me how I can manually authenticate the plugin ?

Or any other way

**Reply by tennisfar** [reply]
Same problem for me, I'm on work laptop, can make Copilot sign in with VS Code (probably because I'm using the plugin "win-ca"), but in Rider it just hangs after I've accepted the "Congratulations, you're all set" dialog.

**Comment by nortain** [comment]
I tried using copilot on both my personal and work systems.  The work system failed to ever connect while my personal system had no issue at all.  It doesn't look like my company has a corporate account so I'll be unlikely to run in on my work system until one is setup and I'm able to use the copilot business account.  Wouldn't be surprised if a lot of folks are the professional work space aren't in a similar place.

**Reply by shibasis0801** [reply]
VS code works for me on the Work system

**Comment by nuisanceless** [comment]
I have same issue。

**Comment by tennisfar** [comment]
I can confirm having the same issue, on work systems.

**Comment by dmitry-js** [comment]
Same here

**Comment by sonu628** [comment]
same issue on Webstorm
UPDATE: fixed by updating

**Reply by Liko2k** [reply]
Can you elaborate please?

**Comment by kuoaid** [comment]
Did you guys go to settings/copilot on the github website to finish the setup? Not doing that WILL block you from using Github Copilot, and the program will not tell you what's wrong.

**Reply by TomVondracek** [reply]
Thank you! Solved by finishing the setup in the https://github.com/settings/copilot

**Comment by benbjamin24** [comment]
Had the same issue, all that was needed was to click the notification button in the VSC and click accept, and that was all.

**Comment by bogdanIuga** [comment]
Had the same issue. It's not working from the copilot modal.

To fix it (worked on my machine :)) ):
1. In VS Code, bottom left, click on the user Icon
2. Sign in to copilot from there

**Comment by zoltan-ongithub** [comment]
Same issue on Android Studio.

**Comment by ZefanHu** [comment]
@gaashh I meet the same issue in Clion, I solved the problem with the following:
first, logout Copilot. then, restart the CLion. And then let Copilot force the proxy use the Proxifier.
Premise, I use the clash for windows to connect  extranet becuase my local network cannot use github smoothly.
Reopen CLion, log in again, and I found that it can be used normally.I may not express very well, my English is not very good. 
I hope this works for you.

**Comment by avishekgurung** [comment]
Has this issue been fixed? I am facing the same in Pycharm 22

**Reply by Liko2k** [reply]
Me too.

**Comment by samcuxx** [comment]
It will Fail to sign in when when you have a bad internet connection

**Reply by Liko2k** [reply]
I am behind the corporate firewall (proxy).

**Comment by va-adh** [comment]
In my corporate network, I have the same problem. 
On Visual Studio Code it works but not on IntelliJ I got "Waiting for GitHub Authentication".

**Comment by v0ry** [comment]
Had the same problem, the problem was I downloaded copilot first from the official webside, but also had a plugin that cofigured copilot which caused the error. If I were you I'd delete copilot, and check if it works afterwords and if not reinstall it.

**Comment by Silbad** [comment]
Hello, same problem, click on "Sign In" to copilot make nothing

**Reply by v0ry** [reply]
`: copilot auth`
wait, window appears. Follow as mentioned

**Reply by v0ry** [reply]
Also I use, it's faster. 

[copilot in lua](https://github.com/zbirenbaum/copilot.lua)

**Reply by v0ry** [reply]
Make sure to delete files  
`rm -rf  ~/.config/github copilort/`
`rm -rf ~/.config/nvim/pack`
So you get rid of the installation from github... Else you will ahe an errro

**Comment by tal19987** [comment]
Still having an issue with VSCode, suddenly stopped working

**Reply by 23875658** [reply]
Just close then reopen your workspace

**Comment by grzla** [comment]
this just started happening here. after i pulled changes to my project from github, now on any project i have, when i start vscode i can see the copilot chat history for a few seconds until everything is loaded, then it disappears and i get 'sign in to github to use copilot, your ai pair programmer.' clicking 'sign in' doesn't do anything. everything is up to date and i've tried restarting the extension.

**Comment by pinyinjj** [comment]
Same issue still exists in 2024

**Reply by pinyinjj** [reply]
nvm issue fixed after using vpn to access copilot auth page
